From maximlevitsky at gmail.com  Fri Feb 29 09:13:28 2008
From: maximlevitsky at gmail.com (Maxim Levitsky)
Date: Fri, 29 Feb 2008 19:13:28 +0200
Subject: [PATCH 2/2] Add an option to turn off textured video
In-Reply-To: <200802291911.08259.maximlevitsky@gmail.com>
References: <200802241815.33057.maximlevitsky@gmail.com>
        <8a7302370802290600w68c13aa8kb43ae82fe003342a@mail.gmail.com>
        <200802291911.08259.maximlevitsky@gmail.com>
Message-ID: <200802291913.28673.maximlevitsky@gmail.com>

From 20416929e91875e9f9a5b46ab142e07e9f9ef27a Mon Sep 17 00:00:00 2001
From: Maxim Levitsky <maximlevitsky at gmail.com>
Date: Fri, 29 Feb 2008 18:29:02 +0200
Subject: [PATCH] Add an option to turn off textured video

Add a Boolean "TexturedVideo" option with default 'false'  value
to be able to turn textured video off.
---
 src/i830.h        |    1 +
 src/i830_driver.c |    6 +++++-
 src/i830_video.c  |    3 ++-
 3 files changed, 8 insertions(+), 2 deletions(-)

[Updated to apply to the 2.3.1 driver -- bryce 6/12/08]

diff -Nurp xserver-xorg-video-intel-2.3.1-1ubuntu1~bwh3-patched/src/i830.h xserver-xorg-video-intel-2.3.1-1ubuntu1~bwh3-working/src/i830.h
--- xserver-xorg-video-intel-2.3.1-1ubuntu1~bwh3-patched/src/i830.h	2008-06-12 16:03:38.000000000 -0700
+++ xserver-xorg-video-intel-2.3.1-1ubuntu1~bwh3-working/src/i830.h	2008-06-12 16:00:34.000000000 -0700
@@ -514,6 +514,7 @@ typedef struct _I830Rec {
 
    Bool XvDisabled;			/* Xv disabled in PreInit. */
    Bool XvEnabled;			/* Xv enabled for this generation. */
+   Bool TexturedXvEnabled;		/* Textured video enabled/disabled by user*/
 
 #ifdef I830_XV
    int colorKey;
diff -Nurp xserver-xorg-video-intel-2.3.1-1ubuntu1~bwh3-patched/src/i830_driver.c xserver-xorg-video-intel-2.3.1-1ubuntu1~bwh3-working/src/i830_driver.c
--- xserver-xorg-video-intel-2.3.1-1ubuntu1~bwh3-patched/src/i830_driver.c	2008-06-12 16:03:38.000000000 -0700
+++ xserver-xorg-video-intel-2.3.1-1ubuntu1~bwh3-working/src/i830_driver.c	2008-06-12 16:00:57.000000000 -0700
@@ -308,6 +308,7 @@ typedef enum {
    OPTION_LVDSFIXEDMODE,
    OPTION_TRIPLEBUFFER,
    OPTION_FORCEENABLEPIPEA,
+   OPTION_TEXTURED_VIDEO,
 #ifdef INTEL_XVMC
    OPTION_XVMC,
 #endif
@@ -336,6 +337,7 @@ static OptionInfoRec I830Options[] = {
    {OPTION_LVDSFIXEDMODE, "LVDSFixedMode", OPTV_BOOLEAN,	{0},	FALSE},
    {OPTION_TRIPLEBUFFER, "TripleBuffer", OPTV_BOOLEAN,	{0},	FALSE},
    {OPTION_FORCEENABLEPIPEA, "ForceEnablePipeA", OPTV_BOOLEAN,	{0},	FALSE},
+   {OPTION_TEXTURED_VIDEO, "TexturedVideo", OPTV_BOOLEAN, {0}, FALSE},
 #ifdef INTEL_XVMC
    {OPTION_XVMC,	"XvMC",		OPTV_BOOLEAN,	{0},	TRUE},
 #endif
@@ -1609,6 +1611,8 @@ I830PreInit(ScrnInfoPtr pScrn, int flags
    pI830->XvDisabled =
 	!xf86ReturnOptValBool(pI830->Options, OPTION_XVIDEO, TRUE);
 
+   pI830->TexturedXvEnabled = OVERLAY_NOEXIST(pI830) || xf86ReturnOptValBool(pI830->Options, OPTION_TEXTURED_VIDEO, FALSE);
+
 #ifdef I830_XV
    if (xf86GetOptValInteger(pI830->Options, OPTION_VIDEO_KEY,
 			    &(pI830->colorKey))) {
diff -Nurp xserver-xorg-video-intel-2.3.1-1ubuntu1~bwh3-patched/src/i830_video.c xserver-xorg-video-intel-2.3.1-1ubuntu1~bwh3-working/src/i830_video.c
--- xserver-xorg-video-intel-2.3.1-1ubuntu1~bwh3-patched/src/i830_video.c	2008-06-12 16:03:38.000000000 -0700
+++ xserver-xorg-video-intel-2.3.1-1ubuntu1~bwh3-working/src/i830_video.c	2008-06-12 16:00:34.000000000 -0700
@@ -606,7 +606,8 @@ I830InitVideo(ScreenPtr pScreen)
     /* Set up textured video if we can do it at this depth and we are on
      * supported hardware.
      */
-    if (pScrn->bitsPerPixel >= 16 && (IS_I9XX(pI830) || IS_I965G(pI830)) &&
+    if (pI830->TexturedXvEnabled && pScrn->bitsPerPixel >= 16
+        && (IS_I9XX(pI830) || IS_I965G(pI830)) &&
 	!(!IS_I965G(pI830) && pScrn->displayWidth > 2048))
     {
 	texturedAdaptor = I830SetupImageVideoTextured(pScreen);
